xgcm/grid.py

```diff
@@ -1255,34 +1258,62 @@ def check_neighbor(link, position):
             self.axes[axis]._facedim = facedim
             self.axes[axis]._connections = axis_links
 
-    def _assign_metrics(self, metrics):
-        """
-        metrics should look like
-           {('X', 'Y'): ['rAC']}
-        check to make sure everything is a valid dimension
-        """
+    # @property
+    # def metrics(self):
+    #   return self._metrics
 
-        self._metrics = {}
+    # @metrics.setter
+    def set_metrics(self, key, value):
 
-        for key, metric_vars in metrics.items():
-            metric_axes = frozenset(_maybe_promote_str_to_list(key))
-            if not all([ma in self.axes for ma in metric_axes]):
+        metric_axes = frozenset(_maybe_promote_str_to_list(key))
+        if not all([ma in self.axes for ma in metric_axes]):
+            raise KeyError(
+                f"Metric axes {metric_axes!r} not compatible with grid axes {tuple(self.axes)!r}"
+            )
```

Comment:
Nitpick: We could invert any/all to simplify comparisons 

```suggestion
        if any(ma not in self.axes for ma in metric_axes):
            raise KeyError(
                f"Metric axes {metric_axes!r} not compatible with grid axes {tuple(self.axes)!r}"
            )
```

Reply:
I like it, but maybe we want to make it even more explicit and loop over the axes? Then we can show which axes are not compatible?
Something like:
```
axes_not_found = []
for ma in metric_axes:
    if ma not in self.axes:
        axes_not_found.append(ma)

if len(axes_not_found) > 0:
    f"Metric axes {axes_not_found!r} not compatible with grid axes {tuple(self.axes)!r}"
```

Just an idea.

Reply:
I'm in favor of the explicit `for loop`. It's easy to understand (for me at least :). We could use a list comprehension if we want to:

```python
axes_not_found = [ma for ma in metric_axes if ma not in self.axes]

if len(axes_not_found) > 0:
    f"Metric axes {axes_not_found!r} not compatible with grid axes {tuple(self.axes)!r}"
```

Reply:
I would leave this up to @jdldeauna. Equally fine with both.

Reply:
I find this for loop easier to understand, too:
```python
axes_not_found = [ma for ma in metric_axes if ma not in self.axes]
```
Thanks!